WoofWare.Myriad.Plugins/ArgParserGenerator.fs

```diff
@@ -1744,23 +1747,19 @@ module internal ArgParserGenerator =
         ]
         |> SynModuleOrNamespace.createNamespace ns
 
-open Myriad.Core
-
 /// Myriad generator that provides a catamorphism for an algebraic data type.
-[<MyriadGenerator("arg-parser")>]
+[<WhippetGenerator>]
 type ArgParserGenerator () =
 
-    interface IMyriadGenerator with
-        member _.ValidInputExtensions = [ ".fs" ]
+    interface IGenerateRawFromRaw with
+        member _.GenerateRawFromRaw (context : RawSourceGenerationArgs) =
```

Comment:
This is the most basic API on which all other APIs are built. I intend adding convenience APIs on top, eventually, but this API shape is how we get the most modular design (which e.g. allows the consumer to freely upgrade Fantomas).